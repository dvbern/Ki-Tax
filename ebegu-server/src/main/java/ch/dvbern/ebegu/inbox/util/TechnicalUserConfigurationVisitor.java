/*
 * Copyright (C) 2024 DV Bern AG, Switzerland
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as
 * published by the Free Software Foundation, either version 3 of the
 * License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <https://www.gnu.org/licenses/>.
 */

package ch.dvbern.ebegu.inbox.util;

import ch.dvbern.ebegu.util.mandant.MandantIdentifier;
import ch.dvbern.ebegu.util.mandant.MandantVisitor;

public class TechnicalUserConfigurationVisitor implements MandantVisitor<TechnicalUserConfiguration> {

	public TechnicalUserConfiguration process(MandantIdentifier mandant) {
		return mandant.accept(this);
	}

	@Override
	public TechnicalUserConfiguration visitBern() {
		return new TechnicalUserConfiguration(
			"88888888-2222-2222-2222-222222222222",
			"99999999-2222-2222-2222-222222222222"
		);
	}

	@Override
	public TechnicalUserConfiguration visitLuzern() {
		return new TechnicalUserConfiguration(
			"88888888-2224-2222-2222-222222222222",
			"99999999-2224-2222-2222-222222222222"
		);	}

	@Override
	public TechnicalUserConfiguration visitSolothurn() {
		return new TechnicalUserConfiguration(
			"88888888-2225-2222-2222-222222222222",
			"99999999-2225-2222-2222-222222222222"
		);
	}

	@Override
	public TechnicalUserConfiguration visitAppenzellAusserrhoden() {
		return new TechnicalUserConfiguration(
			"88888888-2223-2222-2222-222222222222",
			"99999999-2223-2222-2222-222222222222"
		);
	}

	@Override
	public TechnicalUserConfiguration visitSchwyz() {
		return new TechnicalUserConfiguration(
			"88888888-2226-2222-2222-222222222222",
			"99999999-2226-2222-2222-222222222222"
		);
	}
}
