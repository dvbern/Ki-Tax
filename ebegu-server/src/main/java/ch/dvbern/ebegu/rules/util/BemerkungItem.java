/*
 * Copyright (C) 2019 DV Bern AG, Switzerland
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as
 * published by the Free Software Foundation, either version 3 of the
 * License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <https://www.gnu.org/licenses/>.
 */

package ch.dvbern.ebegu.rules.util;

import java.util.Objects;

import javax.annotation.Nonnull;

import ch.dvbern.ebegu.types.DateRange;

public class BemerkungItem implements Comparable<BemerkungItem> {

	private final DateRange range;
	private final String message;

	public BemerkungItem(DateRange range, String message) {
		this.range = range;
		this.message = message;
	}

	@Override
	public boolean equals(Object o) {
		if (this == o) {
			return true;
		}
		if (!(o instanceof BemerkungItem)) {
			return false;
		}
		BemerkungItem that = (BemerkungItem) o;
		return Objects.equals(range, that.range) &&
			Objects.equals(message, that.message);
	}

	@Override
	public int hashCode() {
		return Objects.hash(range, message);
	}

	@Override
	public int compareTo(@Nonnull BemerkungItem otherItem) {
		return this.getRange().compareTo(otherItem.getRange());
	}

	public DateRange getRange() {
		return range;
	}

	public String getMessage() {
		return message;
	}
}
