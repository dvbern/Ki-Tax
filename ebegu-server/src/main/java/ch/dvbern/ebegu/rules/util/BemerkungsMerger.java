/*
 * Ki-Tax: System for the management of external childcare subsidies
 * Copyright (C) 2017 City of Bern Switzerland
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as
 * published by the Free Software Foundation, either version 3 of the
 * License, or (at your option) any later version.
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 * GNU Affero General Public License for more details.
 * You should have received a copy of the GNU Affero General Public License
 * along with this program. If not, see <http://www.gnu.org/licenses/>.
 */

package ch.dvbern.ebegu.rules.util;

import java.time.LocalDate;
import java.util.Collection;
import java.util.Collections;
import java.util.Deque;
import java.util.HashMap;
import java.util.LinkedList;
import java.util.List;
import java.util.Map;
import java.util.SortedSet;
import java.util.StringJoiner;
import java.util.TreeSet;

import javax.annotation.Nullable;

import ch.dvbern.ebegu.entities.VerfuegungZeitabschnitt;
import ch.dvbern.ebegu.types.DateRange;
import ch.dvbern.ebegu.util.Gueltigkeit;
import com.google.common.collect.Multimaps;
import com.google.common.collect.SortedSetMultimap;
import org.apache.commons.lang3.StringUtils;

/**
 * This class is supposed to find the longest timeperiods for which bemerkungen in {@link VerfuegungZeitabschnitt}en exists.
 * It takes a list of {@link VerfuegungZeitabschnitt} that has no overlap and returns a list of every bemerkungtext and its
 * continous ranges
 *
 * Example
 * <pre>
 * |AC  |AB |AC |
 * </pre>
 * where A,B and C are bemerkungen and the | | represent date ranges
 * is turned into to
 * A: 1-3
 * B: 2
 * C: 1,3
 */
public final class BemerkungsMerger {

	private BemerkungsMerger() {
	}

	/**
	 * prints a string in the format "[dateFrom -d dateTo] bemerkungstext\n" for every zeitabschnitt in the list.
	 * It returns a newline separated String
	 */
	@SuppressWarnings("SimplifyStreamApiCallChains")
	@Nullable
	public static String evaluateBemerkungenForVerfuegung(List<VerfuegungZeitabschnitt> zeitabschnitte) {
		if (zeitabschnitte == null || zeitabschnitte.isEmpty()) {
			return null;
		}

		StringJoiner joiner = new StringJoiner("\n");
		Map<String, Collection<DateRange>> rangesByBemerkungKey = evaluateRangesByBemerkungKey(zeitabschnitte);

		for (Map.Entry<String, Collection<DateRange>> stringCollectionEntry : rangesByBemerkungKey.entrySet()) {
			stringCollectionEntry.getValue().stream()
				.forEachOrdered(dateRange -> joiner.add('[' + dateRange.toRangeString() + "] " + stringCollectionEntry.getKey()));
		}

		return joiner.toString();
	}

	/**
	 * analyzes the passed list of zeitabschnitte and finds the longest countinous date-ranges of a unique bemerkung.
	 *
	 * @param zeitabschnitte list to analyze
	 * @return a map with the bemerkung as key and the longest contious ranges as values
	 */
	public static Map<String, Collection<DateRange>> evaluateRangesByBemerkungKey(List<VerfuegungZeitabschnitt> zeitabschnitte) {

		SortedSetMultimap<String, Gueltigkeit> multimap = createMultimap(zeitabschnitte);
		Map<String, Collection<DateRange>> continousRangesPerKey = new HashMap<>();
		multimap.keySet().forEach(bemKey -> {
			Collection<DateRange> contRanges = mergeAdjacentRanges(multimap.get(bemKey));
			continousRangesPerKey.put(bemKey, contRanges);
		});

		return continousRangesPerKey;
	}

	private static Collection<DateRange> mergeAdjacentRanges(@Nullable SortedSet<Gueltigkeit> gueltigkeiten) {
		if (gueltigkeiten == null) {
			return Collections.emptyList();
		}
		Deque<DateRange> rangesWithoutGaps = new LinkedList<>();

		//noinspection SimplifyStreamApiCallChains
		gueltigkeiten.stream()
			.forEachOrdered(gueltigkeit -> {
				if (rangesWithoutGaps.isEmpty()) {
					rangesWithoutGaps.add(new DateRange(gueltigkeit.getGueltigkeit()));
				} else {

					LocalDate lastEndingDate = rangesWithoutGaps.getLast().getGueltigBis();
					//if the periods are adjacent make the existing period longer
					if (lastEndingDate.plusDays(1).equals(gueltigkeit.getGueltigkeit().getGueltigAb())) {
						DateRange longerRange = new DateRange(rangesWithoutGaps.getLast().getGueltigAb(), gueltigkeit.getGueltigkeit().getGueltigBis());
						rangesWithoutGaps.removeLast();
						rangesWithoutGaps.addLast(longerRange);
						//if there is a gap add the new period
					} else if (lastEndingDate.plusDays(1).isBefore(gueltigkeit.getGueltigkeit().getGueltigAb())) {
						rangesWithoutGaps.add(new DateRange(gueltigkeit.getGueltigkeit()));
						//this should not happen since the evaluator is supposed to eliminate gaps
					} else if (lastEndingDate.equals(gueltigkeit.getGueltigkeit().getGueltigAb()) || lastEndingDate.isAfter(gueltigkeit.getGueltigkeit().getGueltigAb())) {
						String message = "The passed list of gueltigkeiten must be ordered and may not have any overlapping gueltigkeiten around date " +
							lastEndingDate + ". The	offending gueltigkeiten are " + rangesWithoutGaps.getLast() + "  and " + gueltigkeit;
						throw new IllegalArgumentException(message);
					}

				}
			});
		return rangesWithoutGaps;
	}

	private static SortedSetMultimap<String, Gueltigkeit> createMultimap(List<VerfuegungZeitabschnitt> zeitabschnitte) {
		SortedSetMultimap<String, Gueltigkeit> multimap = Multimaps.newSortedSetMultimap(new HashMap<>(), () -> new TreeSet<>(Gueltigkeit.GUELTIG_AB_COMPARATOR));
		for (VerfuegungZeitabschnitt verfuegungZeitabschnitt : zeitabschnitte) {
			if (StringUtils.isNotEmpty(verfuegungZeitabschnitt.getBemerkungen())) {
				//hier bemerkungen des zeitabschnitt vorher noch splitten anhand /n

				String[] split = verfuegungZeitabschnitt.getBemerkungen().split("\\n");
				for (String currBemerkung : split) {
					multimap.put(currBemerkung, verfuegungZeitabschnitt);

				}
			}
		}
		return multimap;
	}
}
