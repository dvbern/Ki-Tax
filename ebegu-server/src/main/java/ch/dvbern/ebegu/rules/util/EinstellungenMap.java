package ch.dvbern.ebegu.rules.util;

import java.util.EnumMap;
import java.util.Map;

import javax.annotation.Nonnull;

import ch.dvbern.ebegu.entities.Einstellung;
import ch.dvbern.ebegu.entities.Gesuchsperiode;
import ch.dvbern.ebegu.enums.EinstellungKey;

/**
 * Hilfsklasse fuer das Handling von EinstellungenMaps
 */
public class EinstellungenMap {
	private Map<EinstellungKey, Einstellung> einstellungen = new EnumMap<>(EinstellungKey.class);

	public void addEinstellung(@Nonnull EinstellungKey key, @Nonnull String value, @Nonnull Gesuchsperiode gesuchsperiode) {
		Einstellung einstellung = new Einstellung(
			key, value, gesuchsperiode);
		einstellungen.put(key, einstellung);
	}

	@Nonnull
	public Map<EinstellungKey, Einstellung> getEinstellungen() {
		return einstellungen;
	}
}
