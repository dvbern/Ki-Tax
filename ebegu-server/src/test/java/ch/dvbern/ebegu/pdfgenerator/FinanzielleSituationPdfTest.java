/*
 * Copyright (C) 2024 DV Bern AG, Switzerland
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as
 * published by the Free Software Foundation, either version 3 of the
 * License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <https://www.gnu.org/licenses/>.
 */

package ch.dvbern.ebegu.pdfgenerator;

import java.io.File;
import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Path;
import java.util.Arrays;
import java.util.Collections;

import ch.dvbern.ebegu.entities.Gemeinde;
import ch.dvbern.ebegu.entities.GemeindeStammdaten;
import ch.dvbern.ebegu.entities.Gesuch;
import ch.dvbern.ebegu.entities.Gesuchsperiode;
import ch.dvbern.ebegu.entities.Mandant;
import ch.dvbern.ebegu.entities.Verfuegung;
import ch.dvbern.ebegu.entities.VerfuegungZeitabschnitt;
import ch.dvbern.ebegu.pdfgenerator.finanzielleSituation.FinanzielleSituationPdfGeneratorFactory;
import ch.dvbern.ebegu.test.TestDataUtil;
import ch.dvbern.ebegu.test.mandant.MandantFactory;
import ch.dvbern.ebegu.test.util.TestDataInstitutionStammdatenBuilder;
import ch.dvbern.ebegu.testfaelle.AbstractTestfall;
import ch.dvbern.ebegu.testfaelle.Testfall01_WaeltiDagmar;
import ch.dvbern.ebegu.testfaelle.Testfall02_FeutzYvonne;
import ch.dvbern.ebegu.tests.util.PdfUnitTestUtil;
import ch.dvbern.ebegu.types.DateRange;
import ch.dvbern.ebegu.util.Constants;
import ch.dvbern.ebegu.util.mandant.MandantIdentifier;
import ch.dvbern.lib.invoicegenerator.errors.InvoiceGeneratorException;
import org.apache.commons.io.FileUtils;
import org.hamcrest.Matchers;
import org.junit.jupiter.api.BeforeAll;
import org.junit.jupiter.params.ParameterizedTest;
import org.junit.jupiter.params.provider.EnumSource;
import org.junit.jupiter.params.provider.EnumSource.Mode;

import static org.hamcrest.MatcherAssert.assertThat;

public class FinanzielleSituationPdfTest {

	private static final String PATH_PREFIX = FileUtils.getTempDirectoryPath() + "/kiBon/FinanzielleSituation/";

	@BeforeAll
	static void beforeAll() throws IOException {
		 FileUtils.deleteDirectory(new File(PATH_PREFIX));

		Arrays.stream(MandantIdentifier.values())
			.map(FinanzielleSituationPdfTest::getOutputPath)
			.forEach(path -> {
				//noinspection ResultOfMethodCallIgnored
				path.toFile().mkdirs();
			});
	}

	@ParameterizedTest
	@EnumSource(value = MandantIdentifier.class, mode = Mode.MATCH_ALL)
	void testAlleinstehend(MandantIdentifier identifier) {
		Mandant mandant = MandantFactory.fromIdentifier(identifier);
		Gesuchsperiode gesuchsperiode = TestDataUtil.createGesuchsperiode1718(mandant);
		Gemeinde gemeinde = TestDataUtil.createGemeindeLondon(mandant);

		TestDataInstitutionStammdatenBuilder institution = new TestDataInstitutionStammdatenBuilder(gesuchsperiode);
		var testFall = new Testfall01_WaeltiDagmar(gesuchsperiode, true, gemeinde, institution);
		Gesuch gesuch = setupGesuch(testFall, gesuchsperiode);

		File pdf = generatePdf(gesuch, "WaeltiDagmar.pdf");
		String text = PdfUnitTestUtil.getText(pdf);

		assertThat(text, Matchers.stringContainsInOrder("Berechnung der finanziellen Verhältnisse"));
	}

	@ParameterizedTest
	@EnumSource(value = MandantIdentifier.class, mode = Mode.MATCH_ALL)
	void testVerheirated(MandantIdentifier identifier) {
		Mandant mandant = MandantFactory.fromIdentifier(identifier);
		Gesuchsperiode gesuchsperiode = TestDataUtil.createGesuchsperiode1718(mandant);
		Gemeinde gemeinde = TestDataUtil.createGemeindeLondon(mandant);

		TestDataInstitutionStammdatenBuilder institution = new TestDataInstitutionStammdatenBuilder(gesuchsperiode);
		var testFall = new Testfall02_FeutzYvonne(gesuchsperiode, true, gemeinde, institution);
		Gesuch gesuch = setupGesuch(testFall, gesuchsperiode);

		File pdf = generatePdf(gesuch, "FeutzYvonne.pdf");
		String text = PdfUnitTestUtil.getText(pdf);

		assertThat(text, Matchers.stringContainsInOrder("Berechnung der finanziellen Verhältnisse"));
	}

	private File generatePdf(Gesuch gesuch, String dokumentname) {
		Gesuchsperiode gesuchsperiode = gesuch.getGesuchsperiode();

		// there are no easy-to-use Mandant rules for test cases, so we create our own...
		Verfuegung verfuegung = new Verfuegung();
		VerfuegungZeitabschnitt single =
			new VerfuegungZeitabschnitt(new DateRange(gesuchsperiode.getGueltigkeit().getGueltigAb()).withFullMonths());
		single.getRelevantBgCalculationResult().setEinkommensjahr(gesuchsperiode.getBasisJahr());

		verfuegung.setZeitabschnitte(Collections.singletonList(single));

		GemeindeStammdaten gemeindeStammdaten = TestDataUtil.createGemeindeStammdaten(gesuch.extractGemeinde());
		var generator =
			FinanzielleSituationPdfGeneratorFactory.getGenerator(gesuch, verfuegung, gemeindeStammdaten,
				Constants.START_OF_TIME);
		MandantIdentifier mandant = gesuch.extractMandant().getMandantIdentifier();

		try {
			Path path = Path.of(getOutputPath(mandant).toString(), dokumentname);
			generator.generate(Files.newOutputStream(path));

			return path.toFile();
		} catch (IOException | InvoiceGeneratorException e) {
			throw new IllegalStateException("Failed to generate PDF", e);
		}
	}

	private Gesuch setupGesuch(AbstractTestfall testFall, Gesuchsperiode gesuchsperiode) {
		testFall.createGesuch(gesuchsperiode.getDatumAktiviert());
		testFall.fillInGesuch();

		return testFall.getGesuch();
	}

	static Path getOutputPath(MandantIdentifier mandant) {
		return Path.of(PATH_PREFIX, mandant.name());
	}
}
