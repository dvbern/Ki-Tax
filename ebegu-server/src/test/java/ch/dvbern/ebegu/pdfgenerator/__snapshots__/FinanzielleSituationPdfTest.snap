finsit-testAlleinstehend[APPENZELL_AUSSERRHODEN]=[
Berechnung der finanziellen Verhältnisse (Beilage 1) 
 
Referenznummer 17.000001.002
Einkünfte Dagmar Wälti
Steuerbares Einkommen resp. Reineinkommen (Staatssteuer) 53’265.00
Säule 3a, sofern hauptberuflich BVG versichert 0.00
Säule 3a von Personen, die keiner BVG angehören: Betrag der 10'000 übersteigt 0.00
Einkaufsbeiträge berufliche Vorsorge 0.00
Liegenschaftsaufwand 0.00
Einkünfte BGSA, vereinfacht abgerechnet 0.00
Vorjahresverluste 0.00
Mitgliederbeiträge und Zuwendungen an politische Parteien 0.00
Leistungen an juristische Personen in der Schweiz 0.00
Total Einkünfte 53’265.00
Vermögen Dagmar Wälti
Steuerbares Vermögen 12’147.00
15 % des steuerbaren Vermögens 1’822.05
Massgebendes Einkommen in CHF 55’087.00
Jugendamt
Nussbaumstrasse 21
3006 Bern
Telefon 031 123 12 12
info@bern.ch
www.bern.ch
Bern, <TODAY>
Dagmar Wälti
Testweg 10
3000 Bern
Massgebendes Einkommen
 
Referenznummer 17.000001.002
Name Dagmar Wälti
Von Bis Einkommensänderung Massgebendes Einkommen in CHF
01.08.2017 31.08.2017 0.00

]


finsit-testAlleinstehend[BERN]=[
Berechnung der finanziellen Verhältnisse (Beilage 1) 
 
1 Gemäss Artikel 24 ASIV
2 Gemäss Artikel 24 ASIV
3 Gemäss Artikel 25 ASIV Abs. 1
Referenznummer 17.000001.002
Berechnungsjahr 2016
Einkünfte1 Dagmar Wälti
Nettolohn gemäss Lohnausweis / Steuererklärung 53’265.00
Weitere steuerbare Einkünfte 0.00
Steuerpflichtiges Ersatzeinkommen 0.00
Erhaltene Unterhaltsbeiträge (Alimente) 0.00
Geschäftsgewinn - Durchschnitt der letzten 3 Jahre 0.00
Zwischentotal Einkünfte 53’265.00
Total Einkünfte 53’265.00
Nettovermögen2 Dagmar Wälti
Bruttovermögen 12’147.00
Schulden 0.00
Zwischentotal Nettovermögen 12’147.00
Zwischentotal Nettovermögen insgesamt 12’147.00
5% Nettovermögen 607.00
Abzüge3 Dagmar Wälti
Bezahlte Unterhaltsbeiträge 0.00
Total Abzüge 0.00
Zusammenzug
Total Einkünfte 53’265.00
5% Nettovermögen 607.00
Total Abzüge 0.00
Massgebendes Einkommen vor Abzug für Familiengrösse 53’872.00
Jugendamt
Nussbaumstrasse 21
3006 Bern
Telefon 031 123 12 12
info@bern.ch
www.bern.ch
Bern, <TODAY>
Dagmar Wälti
Testweg 10
3000 Bern
Massgebendes Einkommen
 
Referenznummer 17.000001.002
Name Dagmar Wälti
Von Bis Einkommensjahr Massgebendes Einkommen
vor Abzug der Familiengrösse
in CHF
Familiengrösse Abzug der Familiengrösse in
CHF
Massgebendes Einkommen in
CHF
01.08.2017 31.08.2017 2016 0.00 0.0 0.00 0.00

]


finsit-testAlleinstehend[LUZERN]=[
Berechnung der finanziellen Verhältnisse (Beilage 1) 
 
Referenznummer 17.000001.002
Berechnungsjahr 2016
Massgebendes Einkommen gemäss Selbstdeklaration Dagmar Wälti
Total Einkünfte 53’265.00
Total Abzüge 0.00
Anrechenbares Vermögen gemäss Selbstdeklaration 1’215.00
Massgebendes Einkommen in CHF 54’480.00
Jugendamt
Nussbaumstrasse 21
3006 Bern
Telefon 031 123 12 12
info@bern.ch
www.bern.ch
Bern, <TODAY>
Dagmar Wälti
Testweg 10
3000 Bern
Massgebendes Einkommen
 
Referenznummer 17.000001.002
Name Dagmar Wälti
Von Bis Einkommensjahr Massgebendes Einkommen in CHF
01.08.2017 31.08.2017 2016 0.00

]


finsit-testAlleinstehend[SCHWYZ]=[
Berechnung der finanziellen Verhältnisse (Beilage 1) 
 
Referenznummer 17.000001.002
Gesuchsperiode 2017/2018
Einkünfte Dagmar Wälti
Reineinkommen 53’265.00
Einkäufe berufliche Vorsorge (BVG / 2. Säule) 0.00
Abzüge
Ausserordentlicher Liegenschaftsaufwand 0.00
Vermögen
Reinvermögen 1 12’147.00
Massgebendes Einkommen in CHF 53’265.00
1 Für die Berechnung des massgebenden Einkommens sind 10% des Reinvermögens nach Abzug des Freibetrages von 200'000 CHF
relevant. Falls für das Gesuch zwei Gesuchsteller vorhanden sind und beide eine separate Steuererklärung ausgefüllt haben, wird der
Freibetrag beiden Personen nur anteilsmässig gewährt.
Jugendamt
Nussbaumstrasse 21
3006 Bern
Telefon 031 123 12 12
info@bern.ch
www.bern.ch
Bern, <TODAY>
Dagmar Wälti
Testweg 10
3000 Bern
Massgebendes Einkommen
 
Referenznummer 17.000001.002
Name Dagmar Wälti
Von Bis Massgebendes
Einkommen in CHF
Sozialabzug /
Kinderabzug in CHF
Anspruchsberechtigtes
Einkommen in CHF
01.08.2017 31.08.2017 0.00 0.00 0.00

]


finsit-testAlleinstehend[SOLOTHURN]=[
Berechnung der finanziellen Verhältnisse (Beilage 1) 
 
Referenznummer 17.000001.002
Berechnungsjahr 2016
Einkünfte Dagmar Wälti
Nettolohn gemäss Lohnausweis / Steuererklärung 53’265.00
Abzüge
Totalbetrag für minderjährige oder in beruflicher Ausbildung stehender Kinder 0.00
Entrichtete Unterhaltsbeiträge (Alimente) 0.00
Vermögen
Steuerbares Vermögen 1 12’147.00
Massgebendes Einkommen in CHF 2 53’872.35
1 5% des steuerbaren Vermögens werden zum massgebenden Einkommmen addiert
2 75% des Bruttolohns plus 5% des Nettovermögens ergeben das Massgebende Einkommen
Jugendamt
Nussbaumstrasse 21
3006 Bern
Telefon 031 123 12 12
info@bern.ch
www.bern.ch
Bern, <TODAY>
Dagmar Wälti
Testweg 10
3000 Bern
Massgebendes Einkommen
 
Referenznummer 17.000001.002
Name Dagmar Wälti
Von Bis Einkommensjahr Massgebendes Einkommen in CHF
01.08.2017 31.08.2017 2016 0.00

]


finsit-testVerheirated[APPENZELL_AUSSERRHODEN]=[
Berechnung der finanziellen Verhältnisse (Beilage 1) 
 
Referenznummer 17.000001.002
Einkünfte Yvonne Feutz Tizian Feutz
Steuerbares Einkommen resp. Reineinkommen
(Staatssteuer)
34’022.00 98’521.00
Säule 3a, sofern hauptberuflich BVG versichert 0.00 0.00
Säule 3a von Personen, die keiner BVG angehören: Betrag
der 10'000 übersteigt
0.00 0.00
Einkaufsbeiträge berufliche Vorsorge 0.00 0.00
Liegenschaftsaufwand 0.00 0.00
Einkünfte BGSA, vereinfacht abgerechnet 0.00 0.00
Vorjahresverluste 0.00 0.00
Mitgliederbeiträge und Zuwendungen an politische
Parteien
0.00 0.00
Leistungen an juristische Personen in der Schweiz 0.00 0.00
Zwischentotal Einkünfte 34’022.00 98’521.00
Total Einkünfte 132’543.00
Vermögen Yvonne Feutz Tizian Feutz
Steuerbares Vermögen 96’054.00 0.00
Zwischentotal Steuerbares Vermögen 96’054.00
15 % des steuerbaren Vermögens 14’408.10
Massgebendes Einkommen in CHF 146’951.00
Jugendamt
Nussbaumstrasse 21
3006 Bern
Telefon 031 123 12 12
info@bern.ch
www.bern.ch
Bern, <TODAY>
Yvonne Feutz
Tizian Feutz
Testweg 10
3000 Bern
Massgebendes Einkommen
 
Referenznummer 17.000001.002
Name Yvonne Feutz, Tizian Feutz
Von Bis Einkommensänderung Massgebendes Einkommen in CHF
01.08.2017 31.08.2017 0.00

]


finsit-testVerheirated[BERN]=[
Berechnung der finanziellen Verhältnisse (Beilage 1) 
 
1 Gemäss Artikel 24 ASIV
2 Gemäss Artikel 24 ASIV
3 Gemäss Artikel 25 ASIV Abs. 1
Referenznummer 17.000001.002
Berechnungsjahr 2016
Einkünfte1 Yvonne Feutz Tizian Feutz
Nettolohn gemäss Lohnausweis / Steuererklärung 34’022.00 98’521.00
Weitere steuerbare Einkünfte 0.00 0.00
Steuerpflichtiges Ersatzeinkommen 0.00 0.00
Erhaltene Unterhaltsbeiträge (Alimente) 0.00 0.00
Geschäftsgewinn - Durchschnitt der letzten 3 Jahre 0.00 0.00
Zwischentotal Einkünfte 34’022.00 98’521.00
Total Einkünfte 132’543.00
Nettovermögen2 Yvonne Feutz Tizian Feutz
Bruttovermögen 96’054.00 0.00
Schulden 0.00 0.00
Zwischentotal Nettovermögen 96’054.00 0.00
Zwischentotal Nettovermögen insgesamt 96’054.00
5% Nettovermögen 4’803.00
Abzüge3 Yvonne Feutz Tizian Feutz
Bezahlte Unterhaltsbeiträge 0.00 0.00
Total Abzüge 0.00
Zusammenzug
Total Einkünfte 132’543.00
5% Nettovermögen 4’803.00
Total Abzüge 0.00
Massgebendes Einkommen vor Abzug für Familiengrösse 137’346.00
Jugendamt
Nussbaumstrasse 21
3006 Bern
Telefon 031 123 12 12
info@bern.ch
www.bern.ch
Bern, <TODAY>
Yvonne Feutz
Tizian Feutz
Testweg 10
3000 Bern
Massgebendes Einkommen
 
Referenznummer 17.000001.002
Name Yvonne Feutz, Tizian Feutz
Von Bis Einkommensjahr Massgebendes Einkommen
vor Abzug der Familiengrösse
in CHF
Familiengrösse Abzug der Familiengrösse in
CHF
Massgebendes Einkommen in
CHF
01.08.2017 31.08.2017 2016 0.00 0.0 0.00 0.00

]


finsit-testVerheirated[LUZERN]=[
Berechnung der finanziellen Verhältnisse (Beilage 1) 
 
Referenznummer 17.000001.002
Berechnungsjahr 2016
Massgebendes Einkommen gemäss Selbstdeklaration Yvonne Feutz
& Tizian Feutz
Total Einkünfte 34’022.00
Total Abzüge 0.00
Anrechenbares Vermögen gemäss Selbstdeklaration 9’605.00
Massgebendes Einkommen in CHF 43’627.00
Jugendamt
Nussbaumstrasse 21
3006 Bern
Telefon 031 123 12 12
info@bern.ch
www.bern.ch
Bern, <TODAY>
Yvonne Feutz
Tizian Feutz
Testweg 10
3000 Bern
Massgebendes Einkommen
 
Referenznummer 17.000001.002
Name Yvonne Feutz, Tizian Feutz
Von Bis Einkommensjahr Massgebendes Einkommen in CHF
01.08.2017 31.08.2017 2016 0.00

]


finsit-testVerheirated[SCHWYZ]=[
Berechnung der finanziellen Verhältnisse (Beilage 1) 
 
Referenznummer 17.000001.002
Gesuchsperiode 2017/2018
Einkünfte Yvonne Feutz
& Tizian Feutz
Reineinkommen 34’022.00
Einkäufe berufliche Vorsorge (BVG / 2. Säule) 0.00
Abzüge
Ausserordentlicher Liegenschaftsaufwand 0.00
Vermögen
Reinvermögen 1 96’054.00
Massgebendes Einkommen in CHF 34’022.00
1 Für die Berechnung des massgebenden Einkommens sind 10% des Reinvermögens nach Abzug des Freibetrages von 200'000 CHF
relevant.
Jugendamt
Nussbaumstrasse 21
3006 Bern
Telefon 031 123 12 12
info@bern.ch
www.bern.ch
Bern, <TODAY>
Yvonne Feutz
Tizian Feutz
Testweg 10
3000 Bern
Massgebendes Einkommen
 
Referenznummer 17.000001.002
Name Yvonne Feutz, Tizian Feutz
Von Bis Massgebendes
Einkommen in CHF
Sozialabzug /
Kinderabzug in CHF
Anspruchsberechtigtes
Einkommen in CHF
01.08.2017 31.08.2017 0.00 0.00 0.00

]


finsit-testVerheirated[SOLOTHURN]=[
Berechnung der finanziellen Verhältnisse (Beilage 1) 
 
 
 
Referenznummer 17.000001.002
Berechnungsjahr 2016
Einkünfte Yvonne Feutz
Nettolohn gemäss Lohnausweis / Steuererklärung 34’022.00
Abzüge
Totalbetrag für minderjährige oder in beruflicher Ausbildung stehender Kinder 0.00
Entrichtete Unterhaltsbeiträge (Alimente) 0.00
Vermögen
Steuerbares Vermögen 1 96’054.00
Massgebendes Einkommen in CHF 2 38’824.70
Einkünfte Tizian Feutz
Nettolohn gemäss Lohnausweis / Steuererklärung 98’521.00
Abzüge
Totalbetrag für minderjährige oder in beruflicher Ausbildung stehender Kinder 0.00
Entrichtete Unterhaltsbeiträge (Alimente) 0.00
Vermögen
Steuerbares Vermögen 1 0.00
Massgebendes Einkommen in CHF 2 98’521.00
Zusammenzug
Massgebendes Einkommen Yvonne Feutz 38’824.70
Massgebendes Einkommen Tizian Feutz 98’521.00
Massgebendes Einkommen in CHF 137’345.70
1 5% des steuerbaren Vermögens werden zum massgebenden Einkommmen addiert
2 75% des Bruttolohns plus 5% des Nettovermögens ergeben das Massgebende Einkommen
Jugendamt
Nussbaumstrasse 21
3006 Bern
Telefon 031 123 12 12
info@bern.ch
www.bern.ch
Bern, <TODAY>
Yvonne Feutz
Tizian Feutz
Testweg 10
3000 Bern
Massgebendes Einkommen
 
Referenznummer 17.000001.002
Name Yvonne Feutz, Tizian Feutz
Von Bis Einkommensjahr Massgebendes Einkommen in CHF
01.08.2017 31.08.2017 2016 0.00

]